app/models/user_compliance_info.rb

```diff
@@ -171,6 +200,25 @@ def handle_compliance_info_request
       UserComplianceInfoRequest.handle_new_user_compliance_info(self)
     end
 
+    def clear_guardian_info_if_user_is_18_or_older
+      return unless birthday.present? && birthday <= 18.years.ago
+      return unless saved_change_to_birthday? || birthday_changed?
+
+      self.guardian_first_name = nil
+      self.guardian_last_name = nil
+      self.guardian_email = nil
+      self.guardian_phone = nil
+      self.guardian_street_address = nil
+      self.guardian_city = nil
+      self.guardian_state = nil
+      self.guardian_zip_code = nil
+      self.guardian_date_of_birth = nil
+      self.guardian_tax_id = nil
+      self.guardian_stripe_tos_accepted = false
+      self.guardian_stripe_processing_tos_accepted = false
+      self.guardian_verified = nil
+    end
+
```

Comment:
if a user switches there birthday to an older than 18 date and previously they had guardian details so we clear all of them on save